tools/update-client

```diff
@@ -0,0 +1,25 @@
+#!/bin/sh
+
+# This script creates a new release of the Hypothesis browser extension
+# using the latest version of the client available on npm.
+
+set -eu
+
+LATEST_CLIENT_VERSION=$(npm view hypothesis version)
+
+# Update checkout
+git checkout master
+git fetch
+git reset --hard origin/master
```

Comment:
> `git status --name --untracked-files=no` should output an empty string if there are no uncommitted changes.

Can you also do `git diff --quiet HEAD` and check the exit code? I’ve given it a go and it *seems* to work whether the changes are in the index or not, and ignores untracked files in the same way. I’m aware I may well have missed some case where it fails, though.